### PR TITLE
Add ?automation=true param to functional tests (Fixes #8917)

### LINKED ISSUE #8917: [cro] Implement a way for functional tests to be excluded from experiments
## Description

I'm not sure what a solution looks like right now, but filing this issue as it's been in the back of my head for too long. When we implement traffic cop experiments, such as in https://github.com/mozilla/bedrock/pull/8901, we often have to skip functional tests for a page because the experiment alters the expected functionality. This is more work for the developer, and also loses us test coverage whilst experiments are running.

We should develop a mechanism by which functional tests are excluded from TC experiments. One possible solution could be for tests to use an `automation=true` URL parameter which traffic cop can look for and exclude. There may well be other options worth exploring too.

---

#### :yellow_heart: Success Criteria :yellow_heart:
- [x] Functional tests are excluded from experiments.
- [x] Functionality is documented in A/B testing docs.

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Adds `automation=true` param to functional tests so we can exclude those sessions from experimentation (see https://github.com/mozilla/bedrock/pull/9622)

## Issue / Bugzilla link
#8917

## Testing
Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/211599010

### CODE REVIEW

**Comment by alexgibson** [comment]
@pmac updated as per your suggestion yesterday, thanks

File: tests/pages/regions/join_firefox_form.py [hunk 1]
```diff
@@ -26,5 +30,6 @@ def type_email(self, value):
         self.find_element(*self._email_locator).send_keys(value)
 
     def click_continue(self):
+        self.wait.until(lambda s: self.is_enabled)
```
Comment (alexgibson) [inline tests/pages/regions/join_firefox_form.py:33]: This fixes a flakey test should a page be slow to load.

File: tests/pages/firefox/new/download.py [hunk 2]
```diff
@@ -22,7 +22,10 @@ def is_download_button_displayed(self):
         return self.is_element_displayed(*self._download_button_locator)
 
     def download_firefox(self):
-        self.find_element(*self._download_button_locator).click()
+        link = self.find_element(*self._download_button_locator)
+        href = link.get_attribute('href')
+        self.set_attribute(link, att_name='href', att_value=href + '?automation=true')
```
Comment (alexgibson) [inline tests/pages/firefox/new/download.py:27]: It's not ideal having tests modify an in-page link, but for the purpose what we're doing here I think this is likely OK.

Comment (pmac) [reply tests/pages/firefox/new/download.py:27]: Yeah. Agreed. Can't think of a better way to go about it though.

**Review by pmac — COMMENTED** [review]
This looks great. Nice work ⭐

**Review by pmac — APPROVED** [review]

**Pull request merged**